shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -129,16 +129,33 @@ class JSBackend {
     case Sel(receiver, fieldName) =>
       JSField(translateTerm(receiver), fieldName.name)
     // Turn let into an IIFE.
-    case Let(isRec, Var(name), value, body) =>
+    case Let(true, Var(name), Lam(args, body), expr) =>
```

Comment:
You should probably raise a code-gen error if the thing is not a lambda, instead of falling through to the next case and generating something wrong.

Reply:
But `rhs` (in `Let(isRec: Bool, name: Var, rhs: Term, body: Term)`) might be non-`Lam`s, right? For example, the parse result of `let y = 0 in y` is `Let(false, y, IntLit(0), Var(y))`.

Reply:
Yes, that's the whole point of my comment. Non-`Lam` recursive bindings aren't code-generable.

Reply:
Here's a hint:
```ml
let rec x = { f = x } in x
//│ res: {f: 'a} as 'a
//│ Runtime error:
//│   ReferenceError: x is not defined
```

Reply:
Just fixed that!

Reply:
> Here's a hint:
> 
> ```ocaml
> let rec x = { f = x } in x
> //│ res: {f: 'a} as 'a
> //│ Runtime error:
> //│   ReferenceError: x is not defined
> ```

There is also a scoping issue. It should report: unresolved symbol x. I just fixed it.

Reply:
Great!